### PR TITLE
Add detailed information in top-level README

### PR DESCRIPTION
**Pull request by razvand**
Document repository structure. Add information on developing content and contributing to it. Reference Discord as discussion channel.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+The `ctf/` folder stores CTF challenges.
+There are typically two CTFs: a `final-ctf` and a `mid-ctf`.
+Each challenge has its own folder.
+
+### CTF Challenge Folder Structure
+
+A CTF challenge folder consists of:
+
+* `README.md`: details the challenge name, challenge description, points, hints, vulnerability overview, solution overview, deployment instructions
```
Comment (carrrina) [inline README.md:33]:
```suggestion
* `README.md`: details the challenge name, challenge description, points, hints, vulnerability overview, solution overview, deployment instructions.
```

File: README.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+* `flag`: contains the flag string
```
Comment (carrrina) [inline README.md:34]:
```suggestion
* `flag`: contains the flag string.
```

File: README.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+* `sol/`: stores the reference solution scripts (`solution.py`, `solution.sh`, `solution.c` + `Makefile` or others), and a solution write-up (a detailed set of steps) in `README.md`.
+* `src/` (optional): stores source code and build scripts to generate deployment or public files.
+  To keep things generic, flags shouldn't be hard-coded in source code files.
+  The recommended way is to use a template file and then use a generator script (using `sed` or something similar) that generates the actual file (with the flag inside).
+
+  Most challenges will require a source code folder, but some may not, making this optional.
+* `public/` (optional): stores public files that are to be packed in an archive that is to be made available to participants.
+  They are usually generated from src/ folder, though they may also be static files.
```
Comment (carrrina) [inline README.md:42]:
```suggestion
  They are usually generated from the `src/` folder, though they may also be static files.
```

File: README.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+  This is optional.
+  Some challenges may not have any public archive to be presented to contestants.
+* `deploy/`: stores deployment files, to be used to host the challenge on a remote system to be exploited by participants, typically in the form of a Docker container.
+  It typically consists of the following files:
+
+  * `Dockerfile`: for creating the Docker container that will run the challenge on the remote system.
+  * `docker-compose.yml` / `Makefile`.: configuring the building and running of the Docker containers
```
Comment (carrrina) [inline README.md:50]:
```suggestion
  * `docker-compose.yml` / `Makefile`: configuring the building and running of the Docker containers
```

File: README.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+A CTF challenge folder consists of:
+
+* `README.md`: details the challenge name, challenge description, points, hints, vulnerability overview, solution overview, deployment instructions
+* `flag`: contains the flag string
+* `sol/`: stores the reference solution scripts (`solution.py`, `solution.sh`, `solution.c` + `Makefile` or others), and a solution write-up (a detailed set of steps) in `README.md`.
+* `src/` (optional): stores source code and build scripts to generate deployment or public files.
+  To keep things generic, flags shouldn't be hard-coded in source code files.
+  The recommended way is to use a template file and then use a generator script (using `sed` or something similar) that generates the actual file (with the flag inside).
+
```
Comment (carrrina) [inline README.md:39]: Extra line?

File: README.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+  Most challenges will require a source code folder, but some may not, making this optional.
+* `public/` (optional): stores public files that are to be packed in an archive that is to be made available to participants.
+  They are usually generated from src/ folder, though they may also be static files.
+
```
Comment (carrrina) [inline README.md:43]: Extra line?

File: README.md (hunk 5 above)
Comment (razvand) [reply README.md:39]: Yes, to force a newline in the rendering of Markdown content.

File: README.md (hunk 6 above)
Comment (razvand) [reply README.md:43]: Yes, to force a newline in the rendering of Markdown content.

**Review by gabrielmocanu — APPROVED** [review]

**Review by Adoie — APPROVED** [review]

**Pull request merged**